tests/AbstractFactoryDecoratorTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace byrokrat\id;
+
+use PHPUnit\Framework\TestCase;
+
+class AbstractFactoryDecoratorTest extends TestCase
+{
+    public function testConstruct()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
+            public function createNewInstance(?string $raw): IdInterface {}
+        };
+
+        $this->assertInstanceOf(IdFactory::class, $factory);
+    }
+
+    public function testCreateId()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
+            public function createNewInstance(?string $raw): IdInterface {
+                return new class extends AbstractId {};
+            }
+        };
+
+        $this->assertInstanceOf(AbstractId::class, $factory->createId('123'));
```

Comment:
Same here. Let's go `IdInterface`. Or better still mock `IdInterface` and `assertSame` here..